nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -229,7 +229,7 @@ geometry_msgs::msg::TwistStamped RegulatedPurePursuitController::computeVelocity
   // Find look ahead distance and point on path and publish
   const double lookahead_dist = getLookAheadDistance(speed);
   auto carrot_pose = getLookAheadPoint(lookahead_dist, transformed_plan);
-  carrot_pub_->publish(std::move(createCarrotMsg(carrot_pose)));
```

Comment:
Change this back - we need to be publishing this as a unique pointer.

Reply:
Unfortunately, changing it back will lead to a build error:
```
/Users/runner/ws/src/navigation2/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp:232:24: error: moving a temporary object prevents copy elision [-Werror,-Wpessimizing-move]
  carrot_pub_->publish(std::move(createCarrotMsg(carrot_pose)));
                       ^
/Users/runner/ws/src/navigation2/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp:232:24: note: remove std::move call here
  carrot_pub_->publish(std::move(createCarrotMsg(carrot_pose)));
                       ^~~~~~~~~~                            ~
1 error generated.
```
Full CI Run: https://github.com/nkalupahana/ros2-foxy-macos/runs/2689457060?check_suite_focus=true

Reply:
We publish almost everything using unique pointers, why is this particular publication a problem?

Reply:
This is still publishing with unique pointers -- all I've done here is remove `std::move`, as required by the compiler. This is because in C++17, when compiling with clang, the compiler requires that given certain circumstances (which are met here), the code is optimized to construct the object directly in the caller's stack frame, without ever constructing copies. std::move gets in the way of this.

Here's more information on this from a source that can explain it better than I can; see the `-Wpessimizing-move` section: https://developers.redhat.com/blog/2019/04/12/understanding-when-not-to-stdmove-in-c
GCC should also make this optimization since the C++ standard is set to C++17.

Let me know if this explains the change! TL;DR, removing `std::move` allows the compiler to make a more powerful optimization than `std::move`, and this optimization is required by clang on C++17.

Reply:
OK, if you can submit these changes to `main` and they also build, I'd be OK merging this. I don't want to merge this into only `foxy`  though since it wouldn't apply to any future distributions (and our released distro CI isn't set up for non-member changes)

Reply:
Got it. Since this is the only change that isn't already in `main`, should I make a PR to main with just this change, then?

Reply:
Yeah (you could also do Galactic if you like)

Reply:
Sounds good, I've done `main` and I'll do galactic in a minute.

Reply:
#2395 #2396